src/MLIAP/mliap_so3.cpp

```diff
@@ -0,0 +1,1422 @@
+/* ----------------------------------------------------------------------
+ LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+ https://lammps.sandia.gov/, Sandia National Laboratories
+ Steve Plimpton, sjplimp@sandia.gov
+
+ Copyright (2003) Sandia Corporation.  Under the terms of Contract
+ DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+ certain rights in this software.  This software is distributed under
+ the GNU General Public License.
+
+ See the README file in the top-level LAMMPS directory.
+ ------------------------------------------------------------------------- */
+
+/* ----------------------------------------------------------------------
+ Contributing authors: Aidan Thompson, Christian Trott, SNL
+ ------------------------------------------------------------------------- */
+
+#include "mliap_so3.h"
+#include <cmath>
+#include "math_const.h"
+#include "math_special.h"
+#include "memory.h"
+#include "error.h"
+#include "comm.h"
+
+#ifdef MLIAP_SO3_MKL
```

Comment:
Is there a significant performance advantage from using MKL here?
If yes, then searching for MKL and a more generic define (e.g. `-DLMP_MLIAP_USE_MKL`) should be used otherwise we should just stick with the simpler code path.